akka-actor/src/main/scala/akka/pattern/BackoffOnRestartSupervisor.scala

```diff
@@ -38,7 +38,15 @@ private class BackoffOnRestartSupervisor(
         // to our own Restarts, which involves stopping the child.
         case Restart ⇒
           val childRef = sender()
-          become(waitChildTerminatedBeforeBackoff(childRef) orElse handleBackoff)
+          val nextRestartCount = restartCount + 1
+          if (strategy.maxNrOfRetries >= 0 && nextRestartCount > strategy.maxNrOfRetries) {
+            // If we've exceeded the maximum # of retries allowed by the Strategy, die.
+            log.debug(s"Terminating on restart # $nextRestartCount which exceeds max allowed restarts (${strategy.maxNrOfRetries})")
```

Comment:
Use markers instead (to avoid allocations if debug mode is disabled):
`log.debug(s"Terminating on restart #{} which exceeds max allowed restarts ({})", nextRestartCount, strategy.maxNrOfRetries)`